File: performers/go/perf/docPoolCounter.go
package perf

import (
	"sync"
)

type docPoolCounter struct {
	mu    sync.Mutex
	count int
}

func (c *docPoolCounter) getAndInc() int {
	c.mu.Lock()
	current := c.count
	c.count += 1
	c.mu.Unlock()
	return current
}

func (c *docPoolCounter) resetCounter() {
	c.mu.Lock()
	c.count = 0
	c.mu.Unlock()
}

File: performers/python/doc_pool_counter.py
from threading import Lock

class DocPoolCounter():
    def __init__(self):
        self._lock = Lock()
        self._count = 0
    
    def get_and_inc(self):
        with self._lock:
            current = self._count
            self._count += 1
            return current

    def reset_counter(self):
        with self._lock:
            self._counter = 0
